synfig-studio/src/gui/dialogs/dialog_setup.cpp

```diff
@@ -584,13 +588,46 @@ Dialog_Setup::on_accel_edited(const Glib::ustring& path_string, guint accel_key,
 	iter->set_value(SHORTCUT_COLUMN_ID_ACTION_MODS, accel_mods);
 }
 
-void Dialog_Setup::on_accel_cleared(const Glib::ustring& path_string)
+void
+Dialog_Setup::on_accel_cleared(const Glib::ustring& path_string)
 {
 	auto iter = treeview_accels->get_model()->get_iter(path_string);
 	iter->set_value(SHORTCUT_COLUMN_ID_ACTION_KEY, 0);
 	iter->set_value(SHORTCUT_COLUMN_ID_ACTION_MODS, Gdk::ModifierType(0));
 }
 
+void
+Dialog_Setup::on_restore_default_accels_pressed()
+{
+	std::string message = _("Do you really want to restore the default shortcuts?");
+	bool accepted = App::dialog_message_2b(_("Restore Default Shortcuts"), message, Gtk::MESSAGE_QUESTION, _("Cancel"), _("OK"));
+	if (!accepted)
+		return;
+
+	auto accel_rows = treeview_accels->get_model()->children();
+	auto default_accel_map = App::get_default_accel_map();
+	for (const auto& section_row : accel_rows) {
+		for (auto& row : section_row.children()) {
+			Gtk::AccelKey accel;
+
+			std::string accel_path;
+			row.get_value(SHORTCUT_COLUMN_ID_ACTION_NAME, accel_path);
+
+			bool found = false;
```

Comment:
`found` is set but not used?

Reply:
Ops. It is useless indeed. `Gtk::AccelKey accel` is properly initialized for 'empty' accel. So no problem using it if nothing found in AccelMap

Reply:
Fixed